src/domain/ruleset-repository.ts

```diff
@@ -199,42 +167,6 @@ function validateMetadataTemplate(rulesetRepo: RulesetRepository) {
   }
 }
 
-function validateSourceTemplate(rulesetRepo: RulesetRepository) {
```

Comment:
nit: this PR is really big, does it mix a refactoring with the new feature? If so, it would be kinda nice for reviewers and/or the git blame to have the refactoring separate (maybe next time).

Reply:
It's a refactoring that is directly related to the feature. I needed more functionality around the source template file so the validation just moved into that class.

Reply:
So it could be a prefactoring right? Just pointing out that especially with a busy reviewer you have an easier time landing changes that way.

Reply:
Yeah, I could have gone back to unwind things a bit after I knew what the solution would be. I'll keep it in mind for next time.